dashboard/src/containers/Root.tsx

```diff
@@ -1,55 +1,47 @@
 import Header from "components/Header";
 import Layout from "components/Layout";
 import { ConnectedRouter } from "connected-react-router";
-import * as React from "react";
+import React, { Suspense, useEffect, useState } from "react";
 import { IntlProvider } from "react-intl";
 import { Provider } from "react-redux";
-
 import { getCustomI18nConfig, getDefaulI18nConfig, ISupportedLangs } from "shared/I18n";
 import store, { history } from "../store";
 // TODO(andresmgot): Containers should be no longer needed, replace them when possible
 import ConfigLoaderContainer from "./ConfigLoaderContainer";
 import Routes from "./RoutesContainer";
 
-interface IRootState {
-  locale: string;
-  messages: Record<string, string>;
+async function initLocale() {
+  const fullLang = (navigator.languages && navigator.languages[0]) || navigator.language;
+  const lang = fullLang.toLowerCase().split(/[_-]+/)[0];
+  return await getCustomI18nConfig(ISupportedLangs[lang]);
 }
-class Root extends React.Component<{}, IRootState> {
-  constructor(props: {} | IRootState) {
-    super(props);
-    this.state = getDefaulI18nConfig();
-    this.initLocale();
-  }
 
-  public async initLocale() {
-    const fullLang = (navigator.languages && navigator.languages[0]) || navigator.language;
-    const lang = fullLang.toLowerCase().split(/[_-]+/)[0];
-    this.setState(await getCustomI18nConfig(ISupportedLangs[lang]));
-  }
+function Root() {
+  const [i18nConfig, setI18nConfig] = useState(getDefaulI18nConfig());
+  useEffect(() => {
+    initLocale().then(customI18nConfig => setI18nConfig(customI18nConfig));
+  }, []);
```

Comment:
:+1: 

Now you just need to add a test for this :) You can see an example of how to mock `useEffect` + an async function here: https://github.com/kubeapps/kubeapps/blob/master/dashboard/src/components/Config/AppRepoList/AppRepoList.test.tsx#L75